### ISSUE TITLE
Argument#thenRecursive

### ISSUE THREAD

**Issue by GliczDev** [CommandAPI/CommandAPI#529]
### Description

Well, when using CommandTrees, arguments can sometimes look like that:
```java
.then(new LiteralArgument("arg1")
    .then(new StringArgument("arg2")
        .then(new StringArgument("arg3")
            .then(new DoubleArgument("arg4", 0)
                .then(new StringArgument("arg5"))))))
```
and it looks... bad. (well, it can be a BIG tree)

### Expected code

I suggest that there should be a method called `thenRecursive` in Argument class. It would work just like the code above, however, it would look more like:
```java
.thenRecusvie(
    new LiteralArgument("arg1"),
    new StringArgument("arg2"),
    new StringArgument("arg3"),
    new DoubleArgument("arg4", 0),
    new StringArgument("arg5")
)
```
and it looks much more clean!

**Comment by MC-XiaoHei** [comment]
I've written a working beta version and changed the name to `thenNested` as suggested [in Discord](https://discord.com/channels/745416925924032523/745418987948736543/1304125456005595208), I'm writing the test & documentation and will (hopefully) bring up a pr soon

**Comment by JorelAli** [comment]
Implemented in #613 with a new method `.withNested()` which will be available in the next CommandAPI release.

**Comment by MC-XiaoHei** [comment]
> Implemented in #613 with a new method `.withNested()` which will be available in the next CommandAPI release.

it calls `thenNested()` xD

**Comment by JorelAli** [comment]
Now present in 9.7.0.
